### DISCUSSION TITLE
Do we support multi-signature wallets like safe?

### DISCUSSION THREAD

**Discussion by jasonDesyn** [WalletConnect/walletconnect-monorepo#3339] [Developer Support]
I found that I could not receive the callback from the safe wallet

**Comment by jasonDesyn** [comment]
<img width="712" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/89964643/2f936c3c-d2e6-4507-9da3-18b477948c75">

await provider.request({ method: 'eth_signTypedData', params: [walletAddress, JSON.stringify(unsignedData)] }) 

Why do I report that the eth_signTypedData method does not exist when I use this method to sign?

**Comment by jasonDesyn** [comment]
@glitch-txs

**Comment by glitch-txs** [comment]
Methods and chains are required properties, then you have optionalChains and optionalMethods. If the SCW doesn't support one of the required properties the connection will fail. Just add in chains and methods/events what the SCW supports, you can add the rest on optional namespaces for other wallets. We'll soon push support for empty required namespaces for better support for Smart Contract Wallets, we're currently working on it
